### PR TITLE
Use regex to match exception message from absl::Status

### PR DESCRIPTION
**Pull request by yanavlasov**
Additional Description:
Fix ahead to accommodate future changes in absl::Status which include stack trace in error strings.

Risk Level: Low, Test Only
Testing: Unit Tests
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: N/A

Signed-off-by: Yan Avlasov <yavlasov@google.com>

### CODE REVIEW

File: test/common/router/config_impl_test.cc [hunk 1]
```diff
@@ -1352,7 +1352,7 @@ TEST_F(RouteMatcherTest, TestMatchInvalidInput) {
   NiceMock<Envoy::StreamInfo::MockStreamInfo> stream_info;
   factory_context_.cluster_manager_.initializeClusters(
       {"www2", "root_www2", "www2_staging", "instant-server"}, {});
-  EXPECT_THROW_WITH_MESSAGE(
+  EXPECT_THROW_WITH_REGEX(
```
Comment (wbpcode) [inline test/common/router/config_impl_test.cc:1355]: Is this the only position that  will be affected by the future changes in `absl::Status`?

Comment (yanavlasov) [reply test/common/router/config_impl_test.cc:1355]: Yes, this is the only test affected.

**Review by wbpcode — APPROVED** [review]
Thanks. cc @envoyproxy/envoy-maintainers

**Review by adisuissa — APPROVED** [review]
Thanks, LGTM.

**Pull request merged**